sqlalchemy_models/wallet.py

```diff
@@ -0,0 +1,235 @@
+"""
+SQLAlchemy models for Wallets
+"""
+from __init__ import sa, orm, Base, LedgerAmount
+from ledger import Amount
+import datetime
+
+__all__ = ['Balance', 'Address', 'Credit', 'Debit', 'HWBalance']
+
+
+class Balance(Base):
+    """A user's balance in a single currency. Only the latest record is valid."""
+    id = sa.Column(sa.Integer, sa.Sequence('balance_id_seq'), primary_key=True)
+    total = sa.Column(LedgerAmount, nullable=False)
+    available = sa.Column(LedgerAmount, nullable=False)
+    currency = sa.Column(sa.String(4), nullable=False)  # i.e. BTC, DASH, USD
+    time = sa.Column(sa.DateTime(), default=datetime.datetime.utcnow)
+    reference = sa.Column(sa.String(256), nullable=True)
+
+    # foreign key reference to the owner of this
+    user_id = sa.Column(
+        sa.Integer,
+        sa.ForeignKey('user.id'),
+        nullable=False)
+    user = orm.relationship("User", foreign_keys=[user_id])
+
+    def __init__(self, total, available, currency, reference, user_id):
+        self.total = total
+        self.available = available
+        self.currency = currency
+        self.reference = reference
+        self.user_id = user_id
+        self.load_commodities()
+
+    def __repr__(self):
+        return "<Balance(total=%s, available=%s, currency='%s', reference='%s', user_id=%s, time=%s)>" % (
+                   self.total, self.currency, self.currency,
+                   self.reference, self.user_id, self.time.strftime('%Y/%m/%d %H:%M:%S'))
+
+    @orm.reconstructor
+    def load_commodities(self):
+        """
+        Load the commodities for Amounts in this object.
+        """
+        if isinstance(self.available, Amount):
+            self.available = Amount("{0:.8f} {1}".format(self.available.to_double(), self.currency))
+        else:
+            self.available = Amount("{0:.8f} {1}".format(self.available, self.currency))
+        if isinstance(self.total, Amount):
+            self.total = Amount("{0:.8f} {1}".format(self.total.to_double(), self.currency))
+        else:
+            self.total = Amount("{0:.8f} {1}".format(self.total, self.currency))
+
+
+class Address(Base):
+    """A payment network Address or account number."""
+    id = sa.Column(sa.Integer, sa.Sequence('address_id_seq'), primary_key=True)
+    address = sa.Column(sa.String(64),
+                        nullable=False)  # i.e. 1PkzTWAyfR9yoFw2jptKQ3g6E5nKXPsy8r, 	XhwWxABXPVG5Z3ePyLVA3VixPRkARK6FKy
+    currency = sa.Column(sa.String(4), nullable=False)  # i.e. BTC, DASH, USD
+    network = sa.Column(sa.String(64), nullable=False)  # i.e. Bitcoin, Dash, Crypto Capital
+    state = sa.Column(sa.Enum("pending", "active", "blocked", name='state'), nullable=False)
```

Comment:
I think you want the enum names to be unique, given that they work through type declarations (https://www.postgresql.org/docs/9.1/static/datatype-enum.html).